File: src/MyServiceBus.PostgreSql/PostgreSqlJobOptions.cs
namespace MyServiceBus.Persistence.PostgreSql;

public sealed class PostgreSqlJobOptions
{
    public TimeSpan PollInterval { get; set; } = TimeSpan.FromSeconds(1);

    public TimeSpan LeaseDuration { get; set; } = TimeSpan.FromSeconds(30);

    public TimeSpan HeartbeatInterval { get; set; } = TimeSpan.FromSeconds(5);

    public int BatchSize { get; set; } = 16;

    internal void Validate()
    {
        if (PollInterval <= TimeSpan.Zero)
            throw new InvalidOperationException("The PostgreSQL job poll interval must be greater than zero.");
        if (LeaseDuration <= TimeSpan.Zero)
            throw new InvalidOperationException("The PostgreSQL job lease duration must be greater than zero.");
        if (HeartbeatInterval <= TimeSpan.Zero || HeartbeatInterval >= LeaseDuration)
            throw new InvalidOperationException("The PostgreSQL job heartbeat interval must be greater than zero and shorter than the lease duration.");
        if (BatchSize <= 0)
            throw new InvalidOperationException("The PostgreSQL job batch size must be greater than zero.");
    }
}

File: src/Java/myservicebus-postgresql/src/main/java/com/myservicebus/persistence/postgresql/PostgreSqlJobOptions.java
package com.myservicebus.persistence.postgresql;

import java.time.Duration;

public final class PostgreSqlJobOptions {
    private Duration pollInterval = Duration.ofSeconds(1);
    private Duration leaseDuration = Duration.ofSeconds(30);
    private Duration heartbeatInterval = Duration.ofSeconds(5);
    private int batchSize = 16;

    public Duration getPollInterval() {
        return pollInterval;
    }

    public PostgreSqlJobOptions setPollInterval(Duration value) {
        pollInterval = value;
        return this;
    }

    public Duration getLeaseDuration() {
        return leaseDuration;
    }

    public PostgreSqlJobOptions setLeaseDuration(Duration value) {
        leaseDuration = value;
        return this;
    }

    public Duration getHeartbeatInterval() {
        return heartbeatInterval;
    }

    public PostgreSqlJobOptions setHeartbeatInterval(Duration value) {
        heartbeatInterval = value;
        return this;
    }

    public int getBatchSize() {
        return batchSize;
    }

    public PostgreSqlJobOptions setBatchSize(int value) {
        batchSize = value;
        return this;
    }

    void validate() {
        if (pollInterval == null || pollInterval.isZero() || pollInterval.isNegative()) {
            throw new IllegalStateException("The PostgreSQL job poll interval must be positive");
        }
        if (leaseDuration == null || leaseDuration.isZero() || leaseDuration.isNegative()) {
            throw new IllegalStateException("The PostgreSQL job lease duration must be positive");
        }
        if (heartbeatInterval == null
                || heartbeatInterval.isZero()
                || heartbeatInterval.isNegative()
                || heartbeatInterval.compareTo(leaseDuration) >= 0) {
            throw new IllegalStateException(
                    "The PostgreSQL job heartbeat interval must be positive and shorter than the lease duration");
        }
        if (batchSize <= 0) {
            throw new IllegalStateException("The PostgreSQL job batch size must be positive");
        }
    }
}
